app/src/ui/history/commit-list.tsx

```diff
@@ -43,6 +43,9 @@ interface ICommitListProps {
 
   /** Callback to fire to open a given commit on GitHub */
   readonly onViewCommitOnGitHub: (sha: string) => void
+
+  readonly onCompareListScrolled: (rowNumber: number) => void
```

Comment:
@shiftkey Sounds great. I can add a matching sets of comments to `commit-list`.
Should I add documentation to the rest of the files that this variable and function set are passed through?

Reply:
@shiftkey 
It shouldn't be an issue to make the `scrollTop` value being passed down optional.

If you would like `onCompareListScrolled` to be optional as well I can certainly take a swing at that. Would you like this change just for the `commit-list`, or the equivalent for `changes-list` as well?

Reply:
> It shouldn't be an issue to make the `scrollTop` value being passed down optional.

This can be left as-is as I think the callback isn't really valuable unless you're looking at the `scrollTop` value.

Also, can we rename `rowNumber` to `scrollTop` in here to avoid confusion?

> If you would like `onCompareListScrolled` to be optional as well I can certainly take a swing at that.

The reason why I call this out is there might be a case:

 - you use `CommitList` to render a list of commits
 - but are not interested in maintaining scroll position (it's not required to workaround the mount/unmount events in React)
 - and you have to set this callback anyway because it's not optional

That's a lower priority for me, and I think we can revisit it if there's a case where you encounter this.